api/src/pcapi/core/offers/api.py

```diff
@@ -833,20 +839,21 @@ def edit_stock(
         log_extra={"changes": set(modifications.keys())},
     )
 
+    changes_data = {
+        attr_name: {
+            "old": updated_column_log.old_value,
+            "new": updated_column_log.new_value,
+        }
+        for attr_name, updated_column_log in changes.items()
+    }
     log_extra_data: dict[str, typing.Any] = {
         "offer_id": stock.offerId,
         "stock_id": stock.id,
         "stock_dnBookedQuantity": stock.dnBookedQuantity,
+        "provider_id": editing_provider.id if editing_provider else None,
+        **changes_data,
```

Comment:
Je pense que pour la lisibilité du log ça pourrait être pas mal d'ajouter une clé :
```suggestion
        "changes": changes,
```